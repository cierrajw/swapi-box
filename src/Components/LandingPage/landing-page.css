body{
  background-color: black;
  border: 3px groove #F2B705;
  margin: 1em;
  border-radius: 10px;
  margin: auto;
  /*height: 100%;*/
}

.component-container{
  width: 100%;
  /*height: 100%;*/
}

.right-section{
  display: grid;
  grid-template-rows: .5fr .5fr 5fr;
  grid-template-rows: .1fr .1fr 5fr;
}

.buttons-section{
  margin: 0 auto;
}

.header{
  grid-row: 1;
}

.card-container{
  display: flex;
  justify-content: center;
  grid-row: 3;
}

.card-section{
  text-align: center;
}

/* @media screen and (min-width: 605px) and (max-width: 1113px){

} */

@media screen and (max-width: 1040px){

  .component-container{
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .card-section-button{
    width: 130px;
  }

  .button-selected{
    width: 130px;
  }

  .favorites-button{
    width: 130px;
  }
}

@media screen and (max-width: 1040px){
  .buttons-section{
    width: 40rem;
  }
}

@media screen and (max-width: 679px){

  .card-section-button{
    width: 110px;
    font-size: 1em;
  }

  .swapi-title{
    font-size: 3.5em;
    margin: 1em auto;
  }

  .button-selected{
    font-size: 1em;
    width: 110px;
  }

  .favorites-button{
    font-size: 1em;
    width: 110px;
  }

  .buttons-section{

    width: 30rem;
  }
}

@media screen and (max-width: 527px){

  .buttons-section{
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 18rem;
  margin: 1em auto;
  }

  .swapi-title{
    font-size: 2.5em;
    margin: 1em auto;
  }

  .card-section-button{
    width: 200px;
    font-size: 2em;
  }

  .button-selected{
    font-size: 2em;
    width: 200px;
  }

  .favorites-button{
    font-size: 1em;
    width: 200px;
    font-size: 2em;
  }

  .header-section{
    margin: 1em auto;
  }
}

@media screen and (max-width: 465px){
  .swapi-title{
    font-size: 2em;
    margin: 1em auto;
  }

  .header-section{
    width: 100%;
  }

  .buttons-section{
    width: 90%;
  }

}

@media screen and (max-width: 464px){
  .swapi-title{
    font-size: 2em;
    margin: 1em auto;
  }

  .card-section-button{
    width: 200px;
    font-size: 2em;
  }

  .buttons-section{
    width: 200px;
  }

}
