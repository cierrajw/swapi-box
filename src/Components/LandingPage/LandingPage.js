import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import Header from '../Header/Header';
import ScrollSection from '../ScrollSection/ScrollSection'
import Buttons from '../Buttons/Buttons';
import CardContainer from '../CardContainer/CardContainer';
import './landing-page.css'

class LandingPage extends Component{
  constructor(){
    super();
  }

  render(){

    const { filmText, displayPeopleCards, displayVehicleCards, getPeopleCards, getVehicleCards} = this.props;

    return(
      <div className="LandingPage">
        <main className="component-container">

        <ScrollSection className="scroll-section" filmText={filmText}/>

        <section className="right-section">
          <Header className="header"/>

          <CardContainer className="card-container"
          items={this.props.items}/>

          <Buttons className="buttons-section"
          getPeopleCards={getPeopleCards}
          getVehicleCards={getVehicleCards}
          getPlanetCards={this.props.getPlanetCards}/>
        </section>

        </main>
      </div>
    )
  }
}

export default LandingPage;
