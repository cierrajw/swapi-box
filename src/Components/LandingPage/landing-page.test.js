import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage.js';

describe('Landing Page', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });

});
