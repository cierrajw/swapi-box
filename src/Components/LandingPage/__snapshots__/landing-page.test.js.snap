// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main
        className="component-container"
      >
        <ScrollSection
          className="scroll-section"
        />
        <section
          className="right-section"
        >
          <Header
            className="header"
          />
          <CardContainer
            className="card-container"
          />
          <Buttons
            className="buttons-section"
          />
        </section>
      </main>,
      "className": "LandingPage",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ScrollSection
            className="scroll-section"
          />,
          <section
            className="right-section"
          >
            <Header
              className="header"
            />
            <CardContainer
              className="card-container"
            />
            <Buttons
              className="buttons-section"
            />
          </section>,
        ],
        "className": "component-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "scroll-section",
            "filmText": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                className="header"
              />,
              <CardContainer
                className="card-container"
              />,
              <Buttons
                className="buttons-section"
              />,
            ],
            "className": "right-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "header",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "card-container",
                "items": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "buttons-section",
                "getPeopleCards": undefined,
                "getPlanetCards": undefined,
                "getVehicleCards": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main
          className="component-container"
        >
          <ScrollSection
            className="scroll-section"
          />
          <section
            className="right-section"
          >
            <Header
              className="header"
            />
            <CardContainer
              className="card-container"
            />
            <Buttons
              className="buttons-section"
            />
          </section>
        </main>,
        "className": "LandingPage",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ScrollSection
              className="scroll-section"
            />,
            <section
              className="right-section"
            >
              <Header
                className="header"
              />
              <CardContainer
                className="card-container"
              />
              <Buttons
                className="buttons-section"
              />
            </section>,
          ],
          "className": "component-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "scroll-section",
              "filmText": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Header
                  className="header"
                />,
                <CardContainer
                  className="card-container"
                />,
                <Buttons
                  className="buttons-section"
                />,
              ],
              "className": "right-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "header",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "card-container",
                  "items": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "buttons-section",
                  "getPeopleCards": undefined,
                  "getPlanetCards": undefined,
                  "getVehicleCards": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
        ],
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
